crates/secrets/src/scanner.rs

```diff
@@ -0,0 +1,155 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License, Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+use crate::model::secret_result::{SecretResult, SecretResultMatch};
+use crate::model::secret_rule::SecretRule;
+use anyhow::anyhow;
+use common::analysis_options::AnalysisOptions;
+use common::model::position::{Position, INVALID_POSITION};
+use itertools::Itertools;
+use sds::{RuleConfig, Scanner};
+
+/// Build the SDS scanner used to scan all code using the rules fetched from
+/// our API.
+///
+/// Once the scanner is built, use scanner.scan() to find secrets.
+pub fn build_sds_scanner(rules: &[SecretRule]) -> Scanner {
+    let sds_rules = rules
+        .iter()
+        .map(|r| r.convert_to_sds_ruleconfig())
+        .collect::<Vec<RuleConfig>>();
+    Scanner::new(&sds_rules).unwrap()
```

Comment:
Because the scanner should not fail by construction and our integration tests should catch this. There is no user-defined rule here so it should work.